dotnet/src/Connectors/Connectors.AI.Oobabooga/TextCompletion/OobaboogaTextCompletion.cs

```diff
@@ -0,0 +1,257 @@
+            }
+            else
+            {
+                clientWebSocket = this._webSocket;
+            }
+
+            await clientWebSocket.ConnectAsync(streamingUri.Uri, cancellationToken).ConfigureAwait(false);
+
+            var sendSegment = new ArraySegment<byte>(requestBytes);
+            await clientWebSocket.SendAsync(sendSegment, WebSocketMessageType.Text, true, cancellationToken).ConfigureAwait(false);
+
+            var buffer = new byte[8192]; // Increased buffer size
+            while (true)
+            {
+                MemoryStream messageStream = new();
+                WebSocketReceiveResult result;
+                do
+                {
+                    var segment = new ArraySegment<byte>(buffer);
+                    result = await clientWebSocket.ReceiveAsync(segment, cancellationToken).ConfigureAwait(false);
+                    await messageStream.WriteAsync(buffer, 0, result.Count, cancellationToken).ConfigureAwait(false);
+                } while (!result.EndOfMessage);
+
+                messageStream.Seek(0, SeekOrigin.Begin);
+
+                if (result.MessageType == WebSocketMessageType.Text)
+                {
+                    using var reader = new StreamReader(messageStream, Encoding.UTF8);
+                    var messageText = await reader.ReadToEndAsync().ConfigureAwait(false);
+                    var responseObject = JsonSerializer.Deserialize<TextCompletionStreamingResponse>(messageText);
+
+                    if (responseObject is null)
+                    {
+                        throw new AIException(AIException.ErrorCodes.InvalidResponseContent, "Unexpected response from model")
+                        {
+                            Data = { { "ModelResponse", messageText } },
+                        };
+                    }
+
+                    switch (responseObject.Event)
+                    {
+                        case ResponseObjectTextStreamEvent:
+                            yield return new TextCompletionStreamingResult(responseObject);
+                            break;
+                        case ResponseObjectStreamEndEvent:
+                            await clientWebSocket.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, "", CancellationToken.None).ConfigureAwait(false);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                else if (result.MessageType == WebSocketMessageType.Close)
+                {
+                    await clientWebSocket.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, "", CancellationToken.None).ConfigureAwait(false);
+                }
+
+                if (clientWebSocket.State != WebSocketState.Open)
+                {
+                    break;
+                }
+            }
+        }
+        finally
+        {
+            transientClientWebSocket?.Dispose();
```

Comment:
I'm not sure if creating an instance of ClientWebSocket class and disposing it per method call is good idea because it may lead to port exhaustion issue under high load similar to what may happen if HttpClient instances are created and disposed per request. More details - [link](https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines#pooled-connections).
@stephentoub what do you think?

Reply:
I'll provide a global answer in your next message, since the 2 issues are pretty much related.